superdesk/auth/oauth.py

```diff
@@ -24,53 +24,197 @@
 
 """
 
+import logging
+from datetime import datetime
+from typing import Optional, List
 import superdesk
+from flask import url_for, render_template
+from authlib.integrations.flask_client import OAuth
+from authlib.integrations.requests_client import OAuth2Session
+from authlib.oauth2.rfc6749.wrappers import OAuth2Token
+from bson import ObjectId
+from superdesk.resource import Resource
+from superdesk.services import BaseService
 
-from flask import url_for, session
-from flask_oauthlib.client import OAuth
+from superdesk.auth import auth_user, TEMPLATE
 
-from superdesk.auth import auth_user
 
+logger = logging.getLogger(__name__)
 
 bp = superdesk.Blueprint('oauth', __name__)
+oauth = None
+REFRESH_TOKEN_URL = "https://oauth2.googleapis.com/token"
 
 
 def init_app(app):
+    global oauth
```

Comment:
just wondering why having it global?

Reply:
it's used below in `configure_google` and `refresh_google_token` (like at https://github.com/superdesk/superdesk-core/pull/1941/files/5b3ea4f737544c2dc2ca9b60b0adef8bc0f95375#diff-2498d9983039e50bffec686ea374aca9R216).

I've been thinking about putting everything in a class which would be cleaner, but the flask decorators were not too happy about that, and I've not spent too much time checking what was going wrong, as it's was not bad to stay close to the existing code, to limit the diff.